### ISSUE TITLE
hwloc/base: PMIx not available while get topology

### ISSUE THREAD

**Issue by karasevb** [open-mpi/ompi#5059]
Occurs in v3.1.x (with internal hwloc/pmix components):
```shell
shell$ mpirun -n 2 --map-by dist:span -mca rmaps_dist_device mlx5_1 --report-bindings -mca opal_hwloc_base_verbose 5 hostname
[headnode:26470] hwloc:base:get_topology
[headnode:26470] hwloc:base PMIx not available
[headnode:26470] hwloc:base discovering topology
--------------------------------------------------------------------------
The specified device type cannot be found on at least one node.

  Device: mlx5_1
  Node: node01

Open MPI will map the application by default (BYSLOT).
...
```
This causes the `rmaps/mindist` component gets an error when querying the numa list: https://github.com/open-mpi/ompi/blob/fd6fabe340a2fd88fe24f322babf4afa0c0f3cb0/orte/mca/rmaps/mindist/rmaps_mindist_module.c#L335.

**Comment by karasevb** [comment]
The cause of this issue is not PMIx related. The problem was in getting numa list from hwloc.

**Comment by gpaulsen** [comment]
merged into v4.0.x

**Comment by karasevb** [comment]
Issue solved.
